sdk/ml/azure-ai-ml/azure/ai/ml/_utils/utils.py

```diff
@@ -891,6 +892,22 @@ def get_all_enum_values_iter(enum_type):
         if not key.startswith("_"):
             yield getattr(enum_type, key)
 
+def flat_key_snake_to_pascal(key: str) -> str:
+    key_arr = key.split(".")
+    pascal_array = []
+    for key in key_arr:
+        if key != COMPONENT_DEPLOYMENT:
```

Comment:
It seems weird that we have "app" logic in a util file.

Reply:
Why is this one special? And why does the reverse function (flat_key_pascal_to_snake) not do the reverse of this?

Reply:
If you want to have this kind of logic without breaking the integrity of a "util" fxn you can add like, `exclusion_keys  = None` or some kind of optional `options` fields, or handle this completely outside this util function.

Reply:
I will move this to a different file, however the flat_key_pascal_to_snake and flat_key_snake_to_pascal are the same minus the check for the component_deployment enum. That's because the first level of the flattened data is always "ComponentDeployment" and putting that through the snake_to_pascal function changes the casing